### ISSUE TITLE
Handle translations

### ISSUE THREAD

**Issue by fiammybe** [ImpressCMS/composer-module-installer-plugin#2]
Add the classes to handle translations. This might be a bit more tricky, as there are translations on different levels (core, module, library). Let's start with the module translations.

**Comment by MekDrop** [comment]
Why do you need this? Do you have a plan to switch from language constants to something more powerfull? Like gettext ?

**Comment by fiammybe** [comment]
In the current state, I would like to bundle the translations just like any other addon : with versioning and composer-based updates. I think that will be quite easy to do, and I see it as a 'quick win' :-)

**Comment by MekDrop** [comment]
Ah... than maybe could you than postpone this? I'm thinking about moving from constants to gettext (because it's uses less memory). If everything works out I think we will do not need some separate language repositories because one language - one file. If anyone wants to add new language or do some changes they will do pull requests for the selected module. I think in this case it would be easier for end user to see what translation module have and what not.

**Comment by fiammybe** [comment]
No problem in postponing the translations, but I'd rather wait with the move to gettext until ImpressCMS 2.1, I'd like to go for a release of 2.0 before the end of the year, as long as that's feasible.

**Comment by MekDrop** [comment]
I'll try faster todo something with it. :) The biggest reason why I want this is that I think we could get rid icms_version.php for all modules by using only composer.json for all module data. This would improve speed (in some situations about 50%) when we are dealing with info about modules. And most tricky way of this is how to deal with language constants there. I think gettext could be one of all possibles options.

**Comment by fiammybe** [comment]
Hi @MekDrop I will add the support for installing the different types of packages now, so that this works in it's current state. If changes are needed, they can be implemented on top of a working as-is implementation.

**Comment by MekDrop** [comment]
yeah maybe that good idea. still i'm not sure if is good to use composer for translations :/
